PATCH-P0-FEATURE-DOWNSTREAM: Avoid version for all plugins

Making all storage plugins unversioned.

Maintainer: Michal Hrusecky <Michal.Hrusecky@opensuse.org>

Index: storage/blackhole/Makefile.am
===================================================================
--- storage/blackhole/Makefile.am.orig
+++ storage/blackhole/Makefile.am
@@ -34,7 +34,7 @@ noinst_HEADERS =	ha_blackhole.h
 
 EXTRA_LTLIBRARIES =	ha_blackhole.la
 pkgplugin_LTLIBRARIES =	@plugin_blackhole_shared_target@
-ha_blackhole_la_LDFLAGS=-module -rpath $(pkgplugindir)
+ha_blackhole_la_LDFLAGS=-module -avoid-version -rpath $(pkgplugindir)
 ha_blackhole_la_CXXFLAGS=$(AM_CXXFLAGS) -DMYSQL_DYNAMIC_PLUGIN
 ha_blackhole_la_CFLAGS=	$(AM_CFLAGS) -DMYSQL_DYNAMIC_PLUGIN
 ha_blackhole_la_SOURCES=ha_blackhole.cc
Index: storage/archive/Makefile.am
===================================================================
--- storage/archive/Makefile.am.orig
+++ storage/archive/Makefile.am
@@ -35,7 +35,7 @@ noinst_PROGRAMS	=	archive_test archive_r
 
 EXTRA_LTLIBRARIES =	ha_archive.la
 pkgplugin_LTLIBRARIES =	@plugin_archive_shared_target@
-ha_archive_la_LDFLAGS =	-module -rpath $(pkgplugindir)
+ha_archive_la_LDFLAGS =	-module -avoid-version -rpath $(pkgplugindir)
 ha_archive_la_CXXFLAGS=	$(AM_CXXFLAGS) -DMYSQL_DYNAMIC_PLUGIN
 ha_archive_la_CFLAGS =	$(AM_CFLAGS) -DMYSQL_DYNAMIC_PLUGIN
 ha_archive_la_SOURCES =	ha_archive.cc azio.c
Index: storage/example/Makefile.am
===================================================================
--- storage/example/Makefile.am.orig
+++ storage/example/Makefile.am
@@ -34,7 +34,7 @@ noinst_HEADERS =	ha_example.h
 
 EXTRA_LTLIBRARIES =	ha_example.la
 pkgplugin_LTLIBRARIES =	@plugin_example_shared_target@
-ha_example_la_LDFLAGS =	-module -rpath $(pkgplugindir)
+ha_example_la_LDFLAGS =	-module -avoid-version -rpath $(pkgplugindir)
 ha_example_la_CXXFLAGS=	$(AM_CXXFLAGS) -DMYSQL_DYNAMIC_PLUGIN
 ha_example_la_CFLAGS =	$(AM_CFLAGS) -DMYSQL_DYNAMIC_PLUGIN
 ha_example_la_SOURCES =	ha_example.cc
Index: storage/federated/Makefile.am
===================================================================
--- storage/federated/Makefile.am.orig
+++ storage/federated/Makefile.am
@@ -32,7 +32,7 @@ noinst_HEADERS =	ha_federated.h
 
 EXTRA_LTLIBRARIES =	ha_federated.la
 pkgplugin_LTLIBRARIES =	@plugin_federated_shared_target@
-ha_federated_la_LDFLAGS =	-module -rpath $(pkgplugindir)
+ha_federated_la_LDFLAGS =	-module -avoid-version -rpath $(pkgplugindir)
 ha_federated_la_CXXFLAGS=	$(AM_CXXFLAGS) -DMYSQL_DYNAMIC_PLUGIN
 ha_federated_la_CFLAGS =	$(AM_CFLAGS) -DMYSQL_DYNAMIC_PLUGIN
 ha_federated_la_SOURCES =	ha_federated.cc $(top_srcdir)/mysys/string.c
Index: storage/innodb_plugin/Makefile.am
===================================================================
--- storage/innodb_plugin/Makefile.am.orig
+++ storage/innodb_plugin/Makefile.am
@@ -331,7 +331,7 @@ libinnobase_a_CFLAGS=	$(AM_CFLAGS)
 EXTRA_LTLIBRARIES=	ha_innodb_plugin.la
 pkgplugin_LTLIBRARIES=	@plugin_innodb_plugin_shared_target@
 
-ha_innodb_plugin_la_LDFLAGS=	-module -rpath $(pkgplugindir)
+ha_innodb_plugin_la_LDFLAGS=	-module -avoid-version -rpath $(pkgplugindir)
 ha_innodb_plugin_la_CXXFLAGS=	$(AM_CXXFLAGS) $(INNODB_DYNAMIC_CFLAGS)
 ha_innodb_plugin_la_CFLAGS=	$(AM_CFLAGS) $(INNODB_DYNAMIC_CFLAGS)
 ha_innodb_plugin_la_SOURCES=	$(libinnobase_a_SOURCES)
Index: storage/federatedx/Makefile.am
===================================================================
--- storage/federatedx/Makefile.am.orig
+++ storage/federatedx/Makefile.am
@@ -19,7 +19,7 @@ noinst_HEADERS =	ha_federatedx.h federat
 
 EXTRA_LTLIBRARIES =	ha_federatedx.la
 pkgplugin_LTLIBRARIES =	@plugin_federatedx_shared_target@
-ha_federatedx_la_LDFLAGS =	-module -rpath $(pkgplugindir)
+ha_federatedx_la_LDFLAGS =	-module -avoid-version -rpath $(pkgplugindir)
 ha_federatedx_la_CXXFLAGS=	$(AM_CFLAGS) -DMYSQL_DYNAMIC_PLUGIN
 ha_federatedx_la_CFLAGS =	$(AM_CFLAGS) -DMYSQL_DYNAMIC_PLUGIN
 
